internal_ws/ykcompile/src/arch/x86_64/mod.rs

```diff
@@ -293,9 +293,8 @@ pub extern "sysv64" fn bh_push_vec(
         unsafe { std::str::from_utf8(std::slice::from_raw_parts(sym_ptr, sym_len)).unwrap() };
     let body = SIR.body(fname).unwrap();
     let fi = FrameInfo { body, bbidx, mem };
-    let mut v = unsafe { Box::from_raw(vptr) };
+    let v = unsafe { &mut *vptr };
```

Comment:
Because we convert the pointer into an `&mut`, I do think that we should document (somehow...) that the `mut *` pointer will be converted into an `&mut`, so that someone who's au fait with stacked borrows can realise the implication of calling this function.

Reply:
Something like, note this converts the raw pointer into a mutable reference, but there are no other references for this, so we good on stacked borrows. ;-)

Reply:
I think the doc string should lay out the spec only. So "Note that this function converts the raw pointer into an `&mut` reference." is sufficient. It's up to the caller to satisfy that spec re: stacked borrows.

Reply:
Sorry, I got side-tracked by something else, which I initially wanted to add to this PR, but have decided its better on its own. Try 1b23592.